gemma-core/src/main/java/ubic/gemma/core/ontology/OntologyServiceImpl.java

```diff
@@ -383,6 +380,29 @@ public Collection<CharacteristicValueObject> findTermsInexact( String givenQuery
                 .sorted( getCharacteristicComparator( queryString ) )
                 .collect( Collectors.toList() );
 
+
+        /*
+         * Populate the definition for the top hits.
+         */
+        int numfilled = 0;
+        int maxfilled = 25; // presuming we don't need to look too far down the list ... just as a start.
+        for ( CharacteristicValueObject cvo : sortedResults ) {
+            // FIXME: the definition should be retreivable directly from the OntologyTerm; this is temporary.
+            OntologyTerm ot = this.getTerm( cvo.getValueUri() );
+            if ( ot != null ) {
+                for ( AnnotationProperty ann : ot.getAnnotations() ) {
+                    // FIXME: not clear this will work with all ontologies. UBERON does it this way.
+                    if ( ann.getUri().equals( "http://purl.obolibrary.org/obo/IAO_0000115" ) ) {
+                        cvo.setValueDefinition( ann.getContents() );
+                        break;
+                    }
+                }
+            }
+            if ( ++numfilled > maxfilled ) {
+                break;
+            }
+        }
```

Comment:
You should do that in the `AnnotationController` instead where it's actually needed. It will otherwise penalize other use cases that do not make use of definitions.